src/python/pants/engine/legacy/structs.py

```diff
@@ -351,17 +358,25 @@ def legacy_globs_class(self) -> Type[wrapped_globs.FilesetRelPathWrapper]:
   def __init__(
     self, *patterns: str, spec_path: str, exclude: Optional[List[str]] = None, **kwargs,
   ) -> None:
-    self._patterns = patterns
+    self._include_patterns, self._exclude_patterns = parse_glob_includes_and_excludes(patterns)
     self._spec_path = spec_path
     self._raw_exclude = exclude
+    if isinstance(self._raw_exclude, str):
+      raise ValueError(
+        f'Excludes should be a list of strings. Got: {self._raw_exclude!r}'
+      )
     if kwargs:
       raise ValueError(f'kwargs not supported. Got: {kwargs}')
 
-    self._file_globs = self.legacy_globs_class.to_filespec(patterns).get('globs', [])
+    self._file_globs = self.legacy_globs_class.to_filespec(self._include_patterns).get('globs', [])
     self._excluded_file_globs = self._filespec_for_exclude(
-      exclude or [], spec_path
+      [*self._exclude_patterns, *(exclude or [])], spec_path
```

Comment:
I generally like to put `spec_path` on the next line in these cases, since it's not super clear when reading that there's actually a separate argument after the list on the same line.

Reply:
Sure. I generally follow Black formatting but will make the change + use kwargs to make this more clear.